presto-docs/src/main/sphinx/security/ldap.rst

```diff
@@ -0,0 +1,295 @@
+.. code-block:: none
+
+    authentication.ldap.group-auth-pattern=(&(objectClass=<objectclass_of_user>)(uid=${USER})(memberof=<dn_of_the_authorized_group>))
+
+Example:
+
+.. code-block:: none
+
+    authentication.ldap.group-auth-pattern=(&(objectClass=inetOrgPerson)(uid=${USER})(memberof=CN=AuthorizedGroup,OU=Asia,DC=corp,DC=domain,DC=com))
+
+For OpenLDAP, for this query to work, make sure you enable the
+``memberOf`` `overlay <http://www.openldap.org/doc/admin24/overlays.html>`_.
+
+You can also use this property for scenarios where you want to authorize a user
+based on complex group authorization search queries. For eg: if you want to authorize
+a user belonging to any one of multiple groups (in OpenLDAP),
+then this property can be set as:
+
+.. code-block:: none
+
+    authentication.ldap.group-auth-pattern=(&(|(memberOf=CN=normal_group,DC=corp,DC=com)(memberOf=CN=another_group,DC=com))(objectClass=inetOrgPerson)(uid=${USER}))
+
+.. _cli_ldap:
+
+Presto CLI
+----------
+
+Environment Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TLS Configuration
+~~~~~~~~~~~~~~~~~
+
+Access to the Presto coordinator should be through HTTPS when using LDAP
+authentication. The Presto CLI can use either a :ref:`Java Keystore
+<server_java_keystore>` file or :ref:`Java Truststore <cli_java_truststore>`
+for its TLS configuration.
+
+If you are using keystore file, it can be copied to the client machine and used
+for its TLS configuration. If you are using truststore, you can either use
+default java truststores or create a custom truststore on the CLI. We do not
+recommend using self-signed certificates in production.
+
+Presto CLI Execution
+^^^^^^^^^^^^^^^^^^^^
+
+In addition to the options that are required when connecting to a Presto
+coordinator that does not require LDAP authentication, invoking the CLI
+with LDAP support enabled requires a number of additional command line
+options. You can either use ``--keystore-*`` or ``--truststore-*`` properties
+to secure TLS connection. The simplest way to invoke the CLI is with a
+wrapper script.
+
+.. code-block:: none
+
+    #!/bin/bash
+
+    ./presto \
+    --server https://presto-coordinator.example.com:8443 \
+    --keystore-path /tmp/presto.jks \
+    --keystore-password password \
+    --truststore-path /tmp/presto_truststore.jks
+    --truststore-password password
+    --catalog <catalog> \
+    --schema <schema> \
+    --user <LDAP user>
+    --password
+
+=============================== =========================================================================
+Option                          Description
+=============================== =========================================================================
+``--server``                    The address and port of the Presto coordinator.  The port must
+                                be set to the port the Presto coordinator is listening for HTTPS
+                                connections on. Presto CLI does not support using `http` scheme for
```

Comment:
"http" should have double backquotes (it's really annoying that RST needs two and everything else needs one...)